core/src/main/java/zipkin2/reporter/HttpEndpointSupplier.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2016-2024 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package zipkin2.reporter;
+
+import java.io.Closeable;
+import java.util.List;
+
+/**
+ * HTTP-based {@link BytesMessageSender senders} use this to resolve a potentially-pseudo endpoint
+ * passed by configuration to a real endpoint.
+ *
+ * <h3>Usage Notes</h3>
+ *
+ * <p>{@link BytesMessageSender senders} should implement the following logic:
+ * <ul>
+ *   <li>During build, the sender should invoke the {@linkplain Factory}.</li>
+ *   <li>If the result is {@link Constant}, build the sender to use a static value.</li>
+ *   <li>Otherwise, call {@link HttpEndpointSupplier#get()} each time
+ *       {@linkplain BytesMessageSender#send(List)} is invoked.</li>
+ *   <li>Call {@link #close()} once during {@link BytesMessageSender#close()}.</li>
+ * </ul>
+ *
+ * <h3>Implementation Notes</h3>
+ *
+ * <p>Implement friendly {@code toString()} functions, that include the real endpoint or the one
+ * passed to the {@linkplain Factory}.
+ *
+ * <p>Senders are not called during production requests, rather in time or size bounded loop, in a
+ * separate async reporting thread. Implementations that resolve endpoints via remote calls, such as
+ * from Eureka, should cache internally to avoid blocking the reporter thread on each loop.
+ *
+ * <p>Some senders, such as Armeria, may have more efficient and precise endpoint group logic. In
+ * scenarios where the sender is known, interfaces here may be used as markers. Doing so can satisfy
+ * dependency injection, without limiting an HTTP framework that can handle groups, to a
+ * single-endpoint supplier.
+ *
+ * @since 3.3
+ */
+public interface HttpEndpointSupplier extends Closeable {
+  /**
+   * HTTP {@link BytesMessageSender sender} builders check for this symbol, and return the input as
+   * a {@linkplain Constant} result rather than perform dynamic lookups.
+   *
+   * @since 3.3
+   */
+  Factory CONSTANT_FACTORY = new Factory() {
```

Comment:
Just want to confirm this is fine to be public (private interface statics are post Java 8). Otherwise, a static method constructing a package private class can be less surface

Reply:
yep I added this so there can be a default bean in spring without thinking too much (they replicate our senders)

Reply:
anyway no problem. just was following how like in brave we have constants like this, but it isn't important to continue that.